modules/tinymce/src/themes/silver/main/ts/ui/window/SilverDialogInstanceApi.ts

```diff
@@ -20,11 +21,11 @@ const getCompByName = (access: DialogAccess, name: string): Optional<AlloyCompon
   // while developing (probably), and put it back in for the real thing.
   if (root.getSystem().isConnected()) {
     const form = Composing.getCurrent(access.getFormWrapper()).getOr(access.getFormWrapper());
-    return Form.getField(form, name).fold(() => {
+    return Form.getField(form, name).orThunk(() => {
       const footer = access.getFooter();
-      const footerState = Reflecting.getState(footer);
-      return footerState.get().bind((f) => f.lookupByName(form, name));
-    }, (comp) => Optional.some(comp));
+      const footerState: Optional<FooterState> = Reflecting.getState(footer).get();
+      return footerState.bind((f) => f.lookupByName(name));
+    });
```

Comment:
This here was overly complicated and didn't need to be a fold, so have just cleaned it up. The `lookupByName` function no longer requires a component to be passed since it could have just been using the `footer` component it already had access to (since it's purpose was to find named buttons in the footer).